calculator.c

```diff
@@ -0,0 +1,147 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <ctype.h>
+#include <string.h>
+
+int is_operator(char c)
+{
+    return c == '+' || c == '-' || c == '*' || c == '/';
+}
+
+int operator_precedence(char op)
+{
+    if (op == '*' || op == '/')
+        return 2;
+    if (op == '+' || op == '-')
+        return 1;
+    return 0;
+}
+
+float perform_operation(float a, float b, char op, int *error)
+{
+    switch (op)
+    {
+    case '+':
+        return a + b;
+    case '-':
+        return a - b;
+    case '*':
+        return a * b;
+    case '/':
+        if (b == 0)
+        {
+            *error = 1;
+            return 0;
+        }
+        return a / b;
+    default:
+        *error = 1;
+        return 0;
+    }
+}
+
+void process_operator(float *numbers, int *top_numbers, char *operators, int *top_operators, int *error)
+{
+    if (*top_operators < 0 || *top_numbers < 1)
+    {
+        *error = 1;
+        return;
+    }
+    float b = numbers[(*top_numbers)--];
+    float a = numbers[(*top_numbers)--];
+    char op = operators[(*top_operators)--];
+    numbers[++(*top_numbers)] = perform_operation(a, b, op, error);
+}
+
+float evaluate_expression(const char *expression, int *error)

```

Comment:
use camelCase, everywhere in code

Reply:
I have update all function name with camelCase convention